client/components/title-format-editor/index.jsx

```diff
@@ -0,0 +1,307 @@
+			const currentSelection = editorState.getSelection();
+
+			const tokenEntity = Entity.create( 'TOKEN', 'IMMUTABLE', { name } );
+
+			const contentState = Modifier.replaceText(
+				editorState.getCurrentContent(),
+				currentSelection,
+				title,
+				null,
+				tokenEntity
+			);
+
+			this.updateEditor( EditorState.push(
+				editorState,
+				contentState,
+				'add-token'
+			) );
+		};
+	}
+
+	removeToken( entityKey ) {
+		return () => {
+			const { editorState } = this.state;
+			const currentContent = editorState.getCurrentContent();
+			const currentSelection = editorState.getSelection();
+
+			const block = currentContent.getBlockForKey( currentSelection.focusKey );
+
+			// get characters in entity
+			const indices = block
+				.getCharacterList()
+				.reduce( ( ids, value, key ) => {
+					return entityKey === value.entity
+						? [ ...ids, key ]
+						: ids;
+				}, [] );
+
+			const range = SelectionState
+				.createEmpty( block.key )
+				.set( 'anchorOffset', min( indices ) )
+				.set( 'focusOffset', max( indices ) );
+
+			const withoutToken = EditorState.push(
+				editorState,
+				Modifier.removeRange(
+					currentContent,
+					range,
+					'forward'
+				),
+				'remove-range'
+			);
+
+			const selectionBeforeToken = EditorState.forceSelection(
+				withoutToken,
+				range
+					.set( 'anchorOffset', min( indices ) )
+					.set( 'focusOffset', min( indices ) )
+			);
+
+			this.updateEditor( selectionBeforeToken );
+		};
+	}
+
+	renderTokens( contentBlock, callback ) {
+		contentBlock.findEntityRanges(
+			character => {
+				const entity = character.getEntity();
+
+				if ( null === entity ) {
+					return false;
+				}
+
+				return 'TOKEN' === Entity.get( entity ).getType();
+			},
+			callback
+		);
+	}
+
+	render() {
+		const { editorState } = this.state;
+		const {
+			titleData,
+			translate,
+			tokens,
+			type
+		} = this.props;
+
+		const previewText = type.value
+			? buildSeoTitle( { [ type.value ]: fromEditor( editorState.getCurrentContent() ) }, type.value, titleData )
+			: '';
+
+		const formattedPreview = previewText
+			? `${ translate( 'Preview' ) }: ${ previewText }`
+			: '';
+
+		return (
+			<div className="title-format-editor">
+				<div className="title-format-editor__header">
+					<span className="title-format-editor__title">{ type.label }</span>
+					{ map( tokens, ( title, name ) => (
+						<span
+							key={ name }
+							className="title-format-editor__button"
+							onClick={ this.addToken( title, name ) }
+						>
+							{ title }
+						</span>
+					) ) }
+				</div>
+				<div className="title-format-editor__editor-wrapper">
+					<Editor
+						editorState={ editorState }
+						onChange={ this.updateEditor }
+						ref={ this.storeEditorReference }
+					/>
+				</div>
+				<div className="title-format-editor__preview">{ formattedPreview }</div>
+			</div>
+		);
+	}
+}
+
+TitleFormatEditor.displayName = 'TitleFormatEditor';
+
+TitleFormatEditor.propTypes = {
```

Comment:
Since we're using an es6 class, we can use a static block in the class definition:
```
static propTypes = {
 //...
}
```